Content/Items/UndergroundTemple/Weapons.TempleSpear.cs

```diff
@@ -22,45 +29,389 @@ public override void SetDefaults()
 
-			if (damageDone >= target.life)
-				Projectile.NewProjectile(Projectile.GetSource_FromThis(), target.Center, new Vector2(0, -1), ProjectileType<TempleSpearLight>(), 0, 0);
+			Main.spriteBatch.Draw(starTex, Projectile.Center + new Vector2(100, 0f).RotatedBy(parent.rotation - MathHelper.PiOver2) - Main.screenPosition, null, new Color(150, 150, 10, 0), Projectile.rotation, starTex.Size() / 2f, 0.25f, 0, 0);
+
+			Main.spriteBatch.Draw(starTex, Projectile.Center + new Vector2(100, 0f).RotatedBy(parent.rotation - MathHelper.PiOver2) - Main.screenPosition, null, new Color(255, 255, 255, 0), Projectile.rotation, starTex.Size() / 2f, 0.25f, 0, 0);
+
+			Main.spriteBatch.Draw(bloomTex, Projectile.Center + new Vector2(100, 0f).RotatedBy(parent.rotation - MathHelper.PiOver2) - Main.screenPosition, null, new Color(255, 255, 255, 0) * 0.5f, 0f, bloomTex.Size() / 2f, 1f, 0, 0);
+
+			return false;
+		}
+
+		private void ManageCaches()
+		{
+			cache = new List<Vector2>();
+			for (int i = 0; i < 20; i++)
+			{
+				cache.Add(Vector2.Lerp(Projectile.Center + Projectile.velocity, Projectile.Center + new Vector2(100, 0f).RotatedBy(parent.rotation - MathHelper.PiOver2), i / 20f));
+			}
+
+			cache.Add(Projectile.Center + Projectile.velocity + new Vector2(100, 0f).RotatedBy(parent.rotation - MathHelper.PiOver2));
+		}
+
+		private void ManageTrail()
+		{
+			trail ??= new Trail(Main.instance.GraphicsDevice, 21, new TriangularTip(4), factor => 6f, factor => new Color(255, 255, 100) * TrailFade());
+
+			trail.Positions = cache.ToArray();
+			trail.NextPosition = Projectile.Center + Projectile.velocity + new Vector2(100, 0f).RotatedBy(parent.rotation - MathHelper.PiOver2);
+
+			trail2 ??= new Trail(Main.instance.GraphicsDevice, 21, new TriangularTip(4), factor => 3.5f, factor => new Color(255, 255, 255) * TrailFade());
+
+			trail2.Positions = cache.ToArray();
+			trail2.NextPosition = Projectile.Center + Projectile.velocity + new Vector2(100, 0f).RotatedBy(parent.rotation - MathHelper.PiOver2);
+		}
+
+		private void DrawLaser(SpriteBatch spriteBatch)
+		{
+			spriteBatch.End();
+			Effect effect = Filters.Scene["CeirosRing"].GetShader().Shader;
+
+			Matrix world = Matrix.CreateTranslation(-Main.screenPosition.Vec3());
+			Matrix view = Main.GameViewMatrix.ZoomMatrix;
+			Matrix projection = Matrix.CreateOrthographicOffCenter(0, Main.screenWidth, Main.screenHeight, 0, -1, 1);
+
+			effect.Parameters["time"].SetValue(Main.GameUpdateCount * -0.03f);
+			effect.Parameters["repeats"].SetValue(1);
+			effect.Parameters["transformMatrix"].SetValue(world * view * projection);
+			effect.Parameters["sampleTexture"].SetValue(ModContent.Request<Texture2D>(AssetDirectory.Assets + "GlowTrail").Value);
+			trail?.Render(effect);
+
+			trail2?.Render(effect);
+			effect.Parameters["sampleTexture"].SetValue(ModContent.Request<Texture2D>(AssetDirectory.Assets + "FireTrail").Value);
+			trail?.Render(effect);
```

Comment:
the grouping between linebreaks here is weird -- shouldnt the calls be together seperated by the uniform sets as opposed to whatever this is?